### PR TITLE
202 populate empty database with initial data

### PR DESCRIPTION
**Pull request by eliaryck**
## Models

- **EventPrice**: added `price` FK (`SET_NULL`); updated unique constraint to `(event, price_rank, price)` and index accordingly
- **Location**: all address fields (`street`, `number`, `postal_code`, `city`) are now nullable/blank; improved `__str__` to gracefully omit empty parts
- **Space / Hall**: `space` FK on `Hall` is now nullable; `__str__` now shows `name (city)` format with safe fallback
- **MediaGallery**: `name` is now nullable/blank; `__str__` falls back to `"Unnamed Gallery"`
- **MediaItem**: `gallery` FK is now nullable; `MediaItemType.IMAGE` value changed to `"foto"`; added `OTHER` type; `MediaItemTranslation.link` changed from `URLField` to `CharField`; various `max_length` values increased on `ProductionTranslation` text fields
- **Pricing**: removed `price_min_max_both_null_or_both_set` constraint (minimum/maximum can now be set independently)

## Admin

- **BaseAdmin**: added `list_per_page = 50` and `show_full_result_count = False` defaults
- **All translation admins**: `list_filter` updated from `"language"` to `"language__code"`
- **All admin classes**: added `get_queryset` overrides with `select_related` / `prefetch_related` to eliminate N+1 queries
- **EventAdmin**: added `date_hierarchy`, optimised queryset
- **PriceAdmin**: removed `membership` from `list_filter`
- **ProductionAdmin**: removed `uit_database_theme` from `list_filter`; added `ArtistNameFilter`, `GenreFilter`, `TagFilter`
- **Settings**: added OpenAPI `ApiKey` security scheme

## Scraper

- **Significantly sped up `viernulvier` scraper** — refactored sync logic for improved performance and reduced redundant API calls

## Tests

- Updated all tests to match new model/admin behaviour (constraint names, nullable fields, `language__code` filters, `"foto"` type value)
- Added tests for new `price` FK on `EventPrice` (set-null, uniqueness, indexes)
- Added `BaseAdmin` pagination defaults tests
- Added `_make_progress_callback` and `nee_ja_to_bool` edge-case tests
- Renamed `test_media_library.py` → `test_media_library_models.py`

## Dependencies

- Added `tqdm` to `requirements/base.txt`

### CODE REVIEW

**Review by prince-b-b — APPROVED** [review]
Amazing job! All tests still pass and everything seems to be working as intended.

File: backend/tests/events/test_event_admin.py [hunk 1]
```diff
@@ -54,11 +42,11 @@ def admin_change_url(model, pk):
 
 class TestEventsAdminRegistration(TestCase):
     def test_event_is_registered(self):
-        """Test case for test_event_is_registered."""
+        """Event model moet geregistreerd zijn in de admin."""
```
Comment (Floko2005) [inline backend/tests/events/test_event_admin.py]:
```suggestion
        """Event model has to be registered in admin."""
```

File: backend/tests/events/test_event_admin.py [hunk 2]
```diff
@@ -54,11 +42,11 @@ def admin_change_url(model, pk):
 
 class TestEventsAdminRegistration(TestCase):
     def test_event_is_registered(self):
-        """Test case for test_event_is_registered."""
+        """Event model moet geregistreerd zijn in de admin."""
         self.assertIn(Event, admin.site._registry)
 
     def test_registered_admin_class_for_event(self):
-        """Test case for test_registered_admin_class_for_event."""
+        """Event admin class moet EventAdmin zijn."""
```
Comment (Floko2005) [inline backend/tests/events/test_event_admin.py]:
```suggestion
        """Event admin class has to be of type EventAdmin."""
```

**Review by Floko2005 — CHANGES_REQUESTED** [review]
Everything looks very good! I just saw 2 comments which are in Dutch instead of English.

You should also check the test coverage, it's at 99% but I think you can bring it back to 100!

Very good job, I has no issues with scraping everything from their API :)

**Review by Floko2005 — APPROVED** [review]
